ls1intum/Artemis
# `Programming exercises`: Rename Java project types

### Checklist
#### General
#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
The names of the project types are ambiguous because they only refer to the project type of the template and solution repository and Maven is implicitly the type for the test repository. When [introducing Gradle as a new project type](https://github.com/ls1intum/Artemis/pull/4781), this has to be changed because there also has to be the option for plain template and solution repository with a Gradle test repository. This would interfere with the current project type 'Eclipse'.
In addition, the name 'Eclipse' is not completly correct because the project can also be opened in other IDEs but instead are basically only plain Java.

### Description
Within these changes, the project type 'Eclipse' has been renamed to 'Plain_Maven' and 'Maven' to 'Maven_Maven'. Therefore, the occurrences of these types have been changed as well as the names of the folders where the template exercise repository files are stored in.
However, no database change is required because the DB only saves the index of project type in ProjectType.java, but not an actual name. Since the order has not been changed, this does not cause any problem.
For Java exercises, the form for creating an exercise has separate forms for the project types of template/solution repository and the test repository.
As discussed with @krusche, these changes already include the additional enum values for the Gradle project types to be introduced. The reason behind this is to prevent errors when deploying other PRs after the PR adding Gradle support has been deployed. Those would occur since exercises from this new type and the corresponding enum values would not exist in other deployed PRs.

### Steps for Testing
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises
4. Confirm that old Java exercises are still working.
5. Generate Java programming exercises for the following configurations:
- Project Type: 'Plain Java' and Test Repository Project Type 'Maven'
- Project Type: 'Maven' and Test Repository Project Type 'Maven'.

Play around with the possible configurations (such as package name, SCA, and sequential test runs). Confirm that everything just works as expected.

6. Import a Java programming exercise for both project types and confirm that the repositories and build plans are set up correctly. Check this for both an exercise that has been created before and after this PR has been deployed.

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
\- No change - 

### Screenshots
<img width="890" alt="Bildschirmfoto 2022-03-08 um 15 22 28" src="https://user-images.githubusercontent.com/73833780/157256715-ac0f893a-662d-4099-993e-fb91e2ef51b7.png">
